app/models/user.rb

```diff
@@ -12,22 +12,10 @@ class User < ApplicationRecord
   has_many :posts
   has_many :comments, dependent: :destroy
   has_many :likes, dependent: :destroy
-
-  def friends
-    friends_array = friendships.map { |friendship| friendship.friend if friendship.status }
-    friends_array += inverse_friendships.map { |friendship| friendship.user if friendship.status }
-    friends_array.compact
-  end
-
-  # Users who have yet to confirme friend requests
-  def pending_friends
-    friendships.map { |friendship| friendship.friend unless friendship.status }.compact
-  end
-
-  # Users who have requested to be friends
-  def friend_requests
-    inverse_friendships.map { |friendship| friendship.user unless friendship.status }.compact
-  end
+  has_many :confirmed_friendships, -> { where status: true }, class_name: 'Friendship'
+  has_many :friends, through: :confirmed_friendships
+  has_many :pending_friends, -> { where status: nil }, class_name: 'Friendship'

```

Comment:
Add in this line also the `foreign_key: "friend_id"`
example:
`has_many :pending_friendships, -> { where confirmed: false }, class_name: "Friendship", foreign_key: "user_id"`